scanner/enricher/partial/partial.go

```diff
@@ -0,0 +1,62 @@
+package partial
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/libvuln/driver"
+	"github.com/quay/zlog"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/scanners/scannerv4"
+)
+
+const (
+	// Type is the type of data returned from the Enricher's Enrich method.
+	Type = "message/vnd.stackrox.scannerv4.partial; enricher=partial"
+
+	// This appears above and must be the same.
+	name = "partial"
+)
+
+var (
+	_ driver.Enricher = (*Enricher)(nil)
+)
+
+// Enricher filters out all packages which are not
+// known to be affected by any vulnerabilities.
+//
+// At this time, this is only supported for Node.js packages.
+type Enricher struct{}
+
+// Name implements driver.Enricher and driver.EnrichmentUpdater.
+func (e Enricher) Name() string { return name }
+
+// Enrich returns a list of Node.js package IDs which have no known vulnerabilities.
+func (e Enricher) Enrich(ctx context.Context, _ driver.EnrichmentGetter, vr *claircore.VulnerabilityReport) (string, []json.RawMessage, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "enricher/partial/Enricher.Enrich")
+	// ids tracks the package IDs of invulnerable Node.js packages.
```

Comment:
_/castspell `invulnerable` on package_

Reply:
I'm actually now considering moving this logic into the `mappers` because I don't think this really qualifies as an enrichment to the Claircore data. What do you think?

Reply:
I think this can all just be done in `filterPackages` which I added in this PR.

Since we aren't supposed to touch the vulnerability report in the enricher, I don't think this "partial enricher" really adds anything new to the vulnerabilty report